.github/actions/check_artifact_size/action.yaml

```diff
@@ -0,0 +1,140 @@
+            } else {
+              core.setFailed(`Expected one artifact with name ${{ inputs.name }}. Found ${matchArtifacts.length}.`);
+            }
+
+      - name: 'Unzip artifact from main branch'
+        id: unzip-downloaded-artifact
+        shell: bash
+        run: |
+          unzip "${{ steps.download-artifact.outputs.downloadDir }}/${{ inputs.name }}.zip" -d "${{ steps.download-artifact.outputs.downloadDir }}"
+
+      - name: 'Check new artifact size against main branch'
+        id: check-artifact-size
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+
+            // Store file size thresholds in fileSizeThresholds.
+            const fileSizeThresholds = JSON.parse('${{ inputs.thresholds }}');
+
+            console.log(JSON.stringify(fileSizeThresholds, null, 2));
+
+            for (let file in fileSizeThresholds) {
+              console.log(`Checking file size of ${file}.`);
+
+              const downloadFilePath = path.join('${{ steps.download-artifact.outputs.downloadDir }}', file);
+              if (!fs.existsSync(downloadFilePath)) {
+                console.log(`File ${file} was not uploaded to the main branch.`);
+                continue;
+              }
+
+              const filePath = path.join(process.env.GITHUB_WORKSPACE, '${{ inputs.path }}', file);
+              if(!fs.existsSync(filePath)) {
+                console.log(`File ${filePath} was not created in the current workflow run.`);
+                continue;
+              }
+
+              const oldStats = fs.statSync(downloadFilePath);
+              const oldSize = oldStats.size;
+              const newStats = fs.statSync(filePath);
+              const newSize = newStats.size;
+
+              console.log(`Latest uploaded artifact size on the main branch is ${oldSize / 1024}kB, new artifact size generated in this PR is ${newSize / 1024}kB.`);
+
+              const deltaSize = newSize - oldSize;
+              const deltaThreshold = (Math.abs(deltaSize) / oldSize * 100).toFixed(4);
+
+              if (deltaSize < 0) {
+                console.log(`Artifact size is decreased by ${-1 * deltaSize} (${deltaThreshold}%).`);
+              } else {
+                console.log(`Artifact size is increased by ${deltaSize} (${deltaThreshold}%).`);
+                if (deltaThreshold > fileSizeThresholds[file]) {
+                  const threshold = (fileSizeThresholds[file] * 100).toFixed(4);
+                  console.log(`Artifact size increase exceeds threshold ${threshold}%.`);
+                  core.setOutput("addLabel", true);
+                }
+              }
+            }
+      - name: 'Remove downloaded artifact'
+        id: remove-downloaded-artifact
+        shell: bash
+        run: rm -r "${{ steps.download-artifact.outputs.downloadDir }}"
+      - name: 'Add label for artifact size increase violation'
```

Comment:
Have you verified that this step works? I think it may work on merge _only_. The evergreen workflow has the permission set and I _think_ you actually do get a token with that permission on merge.

```
The fine-grained token must have at least one of the following permission sets:

"Issues" repository permissions (write)
"Pull requests" repository permissions (write)
```

Reply:
No, I was not able to verify this step.
Get the permission error 403 when trying to trigger this step.
I'll add this step to merge.